src/spec-transform-auto-layout.js

```diff
@@ -139,12 +146,20 @@ var getMaxTickLabelSize = function (domainValues, formatter, fnCalcTickLabelSize
         return size;
     }
 
-    var maxXTickText = _.max(domainValues, (x) => formatter(x).toString().length);
+    var maxXTickText;
+    let lastComputed = -Infinity;
```

Comment:
In general I don't like a code which operates with Infinity. It is very hard to read.

I suggest you to replace

_.max by string length

with

sort by string length + take first item